### PR TITLE
fix: Client.getChatById(): Cannot read properties of undefined (reading '_serialized')

### PR DESCRIPTION
**Pull request by guicuton**
# PR Details

Fix undefined `_serialized` param of object `lastReceivedKey` on chat object at *Injected.js* of **WWEBVERSION: 2.2321.6** when use the [Client.getChatById(chatId)](https://docs.wwebjs.dev/Client.html#getChatById) method.

## Description

Added ternary condition on `lastReceivedKey` while is trying to get the chat model.
This param are undefined when you don't have any historic of messages with given contact

## Related Issue

Evaluation error while trying to send message for new chats on **WWEBVERSION: 2.2321.6**
image

## How Has This Been Tested

- [x] Individual accounts
- [x] Business accounts
- [x] Android
- [ ] IOS

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by zvictor — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[![npm](https://img.shields.io/npm/v/whatsapp-web.js.svg)](https://www.npmjs.com/package/whatsapp-web.js) [![Depfu](https://badges.depfu.com/badges/4a65a0de96ece65fdf39e294e0c8dcba/overview.svg)](https://depfu.com/github/pedroslopez/whatsapp-web.js?project_id=9765) ![WhatsApp_Web 2.2306.7](https://img.shields.io/badge/WhatsApp_Web-2.2306.7-brightgreen.svg) [![Discord Chat](https://img.shields.io/discord/698610475432411196.svg?logo=discord)](https://discord.gg/H7DqQs4)  
+[![npm](https://img.shields.io/npm/v/whatsapp-web.js.svg)](https://www.npmjs.com/package/whatsapp-web.js) [![Depfu](https://badges.depfu.com/badges/4a65a0de96ece65fdf39e294e0c8dcba/overview.svg)](https://depfu.com/github/pedroslopez/whatsapp-web.js?project_id=9765) ![WhatsApp_Web 2.2321.6](https://img.shields.io/badge/WhatsApp_Web-2.2321.6-brightgreen.svg) [![Discord Chat](https://img.shields.io/discord/698610475432411196.svg?logo=discord)](https://discord.gg/H7DqQs4)
```
Comment (tuyuribr) [inline README.md:1]: revert this

File: tools/version-checker/.version [hunk 2]
```diff
@@ -1 +1 @@
-2.2306.7
\ No newline at end of file
+2.2321.6
```
Comment (tuyuribr) [inline tools/version-checker/.version:1]: revert this

**Review by tuyuribr — CHANGES_REQUESTED** [review]

**Comment by zvictor** [comment]
In my setup, messages are processed externally (lambda) before a serialized `Message` object is sent back. It then has to load the chat object again, often leading to the error described by the OP.

Since i started using the code provided in this PR, the error is gone (My setup is small, just a couple of messages a day).

File: README.md (hunk 1 above)
Comment (guicuton) [reply README.md]: Done

File: tools/version-checker/.version (hunk 2 above)
Comment (guicuton) [reply tools/version-checker/.version]: Done

**Review by zvictor — APPROVED** [review]

**Review by tofers — APPROVED** [review]

**Comment by CJRMedeiros123** [comment]
Work for me!

**Comment by frreiheit** [comment]
Unfortunately, the problem persists on iOS.

**Comment by Santosl2** [comment]
Up .. Merge this please

**Review by alechkos — APPROVED** [review]

**Review by Santosl2 — APPROVED** [review]

**Review by tuyuribr — APPROVED** [review]

**Review by aliyss — APPROVED** [review]

**Comment by diegofemello** [comment]
Up .. Merge this please²

**Pull request merged**